packages/mui-material/src/IconButton/IconButton.test.js

```diff
@@ -95,6 +95,45 @@ describe('<IconButton />', () => {
     });
   });
 
+  describe('prop: color', () => {
+    it('should render the primary class', () => {
+      const { getByRole } = render(<IconButton color="primary">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorPrimary);
+    });
+    it('should render the secondary class', () => {
+      const { getByRole } = render(<IconButton color="secondary">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorSecondary);
+    });
+    it('should render the error class', () => {
+      const { getByRole } = render(<IconButton color="error">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorError);
+    });
+    it('should render the info class', () => {
+      const { getByRole } = render(<IconButton color="info">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorInfo);
+    });
+    it('should render the success class', () => {
+      const { getByRole } = render(<IconButton color="success">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorSuccess);
+    });
+    it('should render the warning class', () => {
+      const { getByRole } = render(<IconButton color="warning">book</IconButton>);
+      const button = getByRole('button');
+
+      expect(button).to.have.class(classes.colorWarning);
+    });
```

Comment:
I think we can refactor this by using forEach loop iterating on `['primary', 'secondary', 'error', ....]` array containing classes and keep the test small.

Reply:
https://github.com/mui/material-ui/blob/e6698d568fc7794154bde9f4e90cdaf29774cac0/packages/mui-material/src/IconButton/IconButton.test.js#L98-L115

I did what you suggested, except I used an object over an array because I think it's a bit cleaner. Also, do you know why my regression test failed?